syntax = "proto3";

service BookController {
    rpc ListBook(BookListRequest) returns (stream BookResponse) {}
    rpc CreateBook(BookCreateRequest) returns (BookResponse) {}
    rpc RetrieveBook(BookRetrieveRequest) returns (BookResponse) {}
}

message BookResponse {
    int32 book_id = 1;
    string author = 2;
    string country = 3;
    string language = 4;
    string link = 5;
    int32 pages = 6;
    string title = 7;
    int32 year = 8;
}

message BookListRequest {
    int32 book_id = 1;
}

message BookCreateRequest {
    string author = 1;
    string country = 2;
    string language = 3;
    string link = 4;
    int32 pages = 5;
    string title = 6;
    int32 year = 7;
}

message BookRetrieveRequest {
    int32 book_id = 1;
}

